app/Http/routes.php

```diff
@@ -114,6 +114,11 @@
         Route::post('/tracks/delete/{id}', 'Api\Web\TracksController@postDelete');
         Route::post('/tracks/edit/{id}', 'Api\Web\TracksController@postEdit');
 
+        Route::post('/tracks/{id}/version-upload', 'Api\Web\TracksController@postUploadNewVersion');
+        Route::get('/tracks/{id}/version-change/{version}', 'Api\Web\TracksController@getChangeVersion');
+        Route::get('/tracks/{id}/version-upload-status', 'Api\Web\TracksController@getVersionUploadStatus');
```

Comment:
There's a lot of apparent similarity between uploading versions and the initial file. Have you considered collapsing them into the same endpoint (ie. so that the initial upload comes in through the version endpoint as "version 1" or something)?